Update to HW Accel to show how to pass devices and update Rpi4 Information (#183)
jellyfin/jellyfin-docs: general/administration/hardware-acceleration.md

# Hardware Acceleration

Jellyfin supports [hardware acceleration](https://trac.ffmpeg.org/wiki/HWAccelIntro) (HWA) of video encoding/decoding using FFMpeg. FFMpeg and Jellyfin can support multiple hardware acceleration implementations such as Intel Quicksync (QSV), AMD AMF, nVidia NVENC/NVDEC, OpenMax OMX, and MediaCodec through the Video Acceleration API (VAAPI).

[VAAPI](https://en.wikipedia.org/wiki/Video_Acceleration_API) is a VA-API (video accelerated API) that uses [libva](https://github.com/intel/libva/blob/master/README.md) to interface with local drivers to provide HWA. [QSV](https://trac.ffmpeg.org/wiki/Hardware/QuickSync) uses a modified (forked) version of VAAPI and interfaces it with [libmfx](https://github.com/intel/media-driver/blob/master/README.md), their proprietary drivers [(list of supported processors for QSV)](https://ark.intel.com/content/www/us/en/ark.html#@Processors).

OS | Recommended HW Acceleration
------------ | -------------

> Linux/GNU | QSV, NVENC, VAAPI 

Reviewer: ```suggestion
Linux/GNU | QSV, NVENC, VAAPI
```

# Hardware Acceleration

> Jellyfin supports [hardware acceleration](https://trac.ffmpeg.org/wiki/HWAccelIntro) (HWA) of video encoding/decoding using FFMpeg. FFMpeg and Jellyfin can support multiple hardware acceleration implementations such as Intel Quicksync (QSV), AMD AMF, nVidia NVENC/NVDEC, OpenMax OMX, and MediaCodec through the Video Acceleration API (VAAPI).

Reviewer 2: ```suggestion
Jellyfin supports [hardware acceleration](https://trac.ffmpeg.org/wiki/HWAccelIntro) (HWA) of video encoding/decoding using FFMpeg. FFMpeg and Jellyfin can support multiple hardware acceleration implementations such as Intel Quicksync (QSV), AMD AMF, nVidia NVENC/NVDEC, OpenMax OMX and MediaCodec through the Video Acceleration API (VAAPI).
```

Author: I like my oxford comma D: lol

> [VAAPI](https://en.wikipedia.org/wiki/Video_Acceleration_API) is a VA-API (video accelerated API) that uses [libva](https://github.com/intel/libva/blob/master/README.md) to interface with local drivers to provide HWA. [QSV](https://trac.ffmpeg.org/wiki/Hardware/QuickSync) uses a modified (forked) version of VAAPI and interfaces it with [libmfx](https://github.com/intel/media-driver/blob/master/README.md), their proprietary drivers [(list of supported processors for QSV)](https://ark.intel.com/content/www/us/en/ark.html#@Processors).

Reviewer 2: ```suggestion
[VAAPI](https://en.wikipedia.org/wiki/Video_Acceleration_API) is a Video Acceleration API that uses [libva](https://github.com/intel/libva/blob/master/README.md) to interface with local drivers to provide HWA. [QSV](https://trac.ffmpeg.org/wiki/Hardware/QuickSync) uses a modified (forked) version of VAAPI and interfaces it with [libmfx](https://github.com/intel/media-driver/blob/master/README.md) and their proprietary drivers [(list of supported processors for QSV)](https://ark.intel.com/content/www/us/en/ark.html#@Processors).
```

OS | Recommended HW Acceleration
------------ | -------------
Linux/GNU | QSV, NVENC, VAAPI
Windows | QSV, NVENC, AMF, VAAPI
MacOS | None (videotoolbox support coming)
Android | MediaCodec, OMX
RPi | OMX

[Graphics Cards comparison using HWA](https://www.elpamsoft.com/?p=Plex-Hardware-Transcoding)

[NVIDIA using ffmpeg official list](https://developer.nvidia.com/ffmpeg). Not every card has been tested. These [drivers](https://github.com/keylase/nvidia-patch) are recommended for Linux/GNU and Windows. Here is the official list of [NVIDIA Graphics Cards](https://developer.nvidia.com/video-encode-decode-gpu-support-matrix) for supported codecs. Example of Ubuntu working with [NVENC](https://www.reddit.com/r/jellyfin/comments/amuyba/nvenc_nvdec_working_in_jellyfin_on_ubuntu_server/).

List of supported codecs for [VAAPI](https://wiki.archlinux.org/index.php/Hardware_video_acceleration#Comparison_tables).

AMF Linux Support still [not official](https://github.com/GPUOpen-LibrariesAndSDKs/AMF/issues/4) and AMD GFX Cards are required to use VAAPI on linux.

> Zen is CPU only. No hardware acceleration for any form of video decoding/encoding. You need APU or dGPU. It will be a while before AV1 hardware acceleration shows up though (realistically 2020+).

Reviewer 2: ```suggestion
Zen is CPU only. No hardware acceleration for any form of video decoding/encoding. You need an APU or dGPU.
```

Intel QSV Benchmarks on [Linux](https://www.intel.com/content/www/us/en/cloud-computing/cloud-computing-quicksync-video-ffmpeg-white-paper.html)

> On Windows you can use DXVA2/D3D11VA libraries for decoding instead of libmfx, and HWA encoding on Windows requires libmfx. The DXVA2/D3D11VA libraries are currently not part of Jellyfin. 

Reviewer 2: ```suggestion
On Windows you can use DXVA2/D3D11VA libraries for decoding instead of libmfx and HWA encoding on Windows requires libmfx. The DXVA2/D3D11VA libraries are currently not supported by of Jellyfin. 
```

Each hardware acceleration type, as well as each Jellyfin installation type, requires different setup options before it can be used. It is always best to consult the FFMpeg documentation on the acceleration type you choose for the latest information.

### Acceleration on Docker

In order to use Hardware acceleration in Docker, the devices must be passed to the container. To see what video devices are available, you can run `sudo lshw -c video` or `vainfo`

Note: [NVIDIA GPU's](https://github.com/docker/compose/issues/6691) currently aren't supported in Docker-compose. 

 Docker run configuration example:
 

>    `docker run -d \`  

Reviewer: I would put this in block quotes like the other sections.

Author: I copied it from the installing page to make them match. I could fix both pages or leave it for now lol

    `--volume /path/to/config:/config \`  
    `--volume /path/to/cache:/cache \`  
    `--volume /path/to/media:/media \`  
    `--net=host \`  
    `--restart=unless-stopped \`  
    `--device /dev/dri/renderD128:/dev/dri/renderD128 \`  
    `--device /dev/dri/card0:/dev/dri/card0 \`  
    `jellyfin/jellyfin`
  
Alternatively, using docker-compose:  

> ```

Reviewer 2: ```suggestion
```yaml
```
